### PR TITLE
Add URL parameter reference tests

### PR DESCRIPTION
**Pull request by SlayterDev**
This PR adds reference tests for URL parameter stripping

### CODE REVIEW

File: amp-protections/config_reference.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "readme": "This configuration has only tracking link features",
```
Comment (kdzwinel) [inline amp-protections/config_reference.json:2]: 👍

File: tracker-radar-tests/TR-domain-matching/domain_matching_tests.json [hunk 2]
```diff
@@ -81,7 +81,10 @@
                 "siteURL": "https://randomsite123.com/",
                 "requestURL": "https://bad.cnames.test/breakage",
                 "requestType": "script",
-                "expectAction": "ignore"
+                "expectAction": "ignore",
+                "exceptPlatforms": [
+                    "ios-browser"
```
Comment (kdzwinel) [inline tracker-radar-tests/TR-domain-matching/domain_matching_tests.json:86]: Oh, why those tests are suddenly failing?

File: url-parameters/tests.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            {
+                "name": "Retain all non tracking parameters",
+                "testURL": "http://www.example.com/test.html?q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Don't remove parameters because domain is in exceptions list",
+                "testURL": "http://example2.com/test.html?gclid=test&fbclid=test",
+                "expectURL": "",
```
Comment (kdzwinel) [inline url-parameters/tests.json:28]: this doesn't seem right - should be same URL looking at the test above, right?

File: url-parameters/tests.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Remove 2 tracking parameters and retain other parameters",
+                "testURL": "http://www.example.com/test.html?utm_source=test&utm_medium=test&q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            }
```
Comment (kdzwinel) [inline url-parameters/tests.json:36]: some more edge-cases:

1. param is in the domain name/path like `http://gclid.test.com/fbclid=123`
2. param has no value like `https://some.test.com?fbclid&gclid` (not actually sure what should happen in this case?)
3. params is ALMOST a match like `https://some.test.com?fbclid1=123`

**Review by kdzwinel — COMMENTED** [review]
Param stripping tests look good, nice work! I came up with some more edge-cases and was wondering about those ignored tracker blocking tests?

File: tracker-radar-tests/TR-domain-matching/domain_matching_tests.json (hunk 2 above)
Comment (SlayterDev) [reply tracker-radar-tests/TR-domain-matching/domain_matching_tests.json:86]: I don't think these test cases were tests on iOS when they were added (btw fixed this. There are only 2 new failing tests). I need to sync up with someone on these tests. Idk if there is a regression in iOS or if these are testing something we aren't supporting or a wrong implementation. Regardless I think fixing that is out of scope for this project at the moment but its blocking me with test failures.

File: url-parameters/tests.json (hunk 3 above)
Comment (SlayterDev) [reply url-parameters/tests.json:28]: I could make an exception in the platform test implementation but the current implementation of the feature returns nil if there is no change to the URL.

Comment (SlayterDev) [reply url-parameters/tests.json:28]: Ok slight correction. If a URL is excluded (i.e. user exception, config exception) it returns nil. If a URL is not excluded but not rewritten we get the original URL.

File: tracker-radar-tests/TR-domain-matching/domain_matching_tests.json (hunk 2 above)
Comment (kdzwinel) [reply tracker-radar-tests/TR-domain-matching/domain_matching_tests.json:86]: Yeah agreed, this is an interesting scenario that I'll document. Related fix tasks for those failing tests is https://app.asana.com/0/1200194497630846/1201531896279377/f

File: url-parameters/tests.json (hunk 3 above)
Comment (kdzwinel) [reply url-parameters/tests.json:28]: Gotcha, but this is very iOS specific and other implementation can return unchanged URL in this case. What we expect is the URL to be exactly the same so I think `expectURL` should reflect that. Instead of having empty string here, you should change iOS integration to sth like:

```js

    $cleanURL = LinkCleaner.removeTrackingParameters($testURL)

    if ($cleanURL === nil) { // url unchanged
       $cleanURL = $testURL;
    }

     expect($cleanURL === $test.expectURL)

**Review by kdzwinel — APPROVED** [review]
Left one last comment, fell free to merge after you address it. Thanks for this!
